test/optimizer/cost_test.cpp

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// cost_test.cpp
+//
+// Identification: test/optimizer/cost_test.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "common/harness.h"
+
+#define private public
+
+#include <vector>
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "common/logger.h"
+#include "optimizer/stats/cost.h"
+#include "optimizer/stats/table_stats.h"
+#include "optimizer/stats/value_condition.h"
+#include "optimizer/stats/stats_storage.h"
+#include "optimizer/stats/cost.h"
+#include "type/types.h"
+#include "type/value_factory.h"
+#include "executor/testing_executor_util.h"
+#include "catalog/column_catalog.h"
+#include "catalog/catalog.h"
+#include "sql/testing_sql_util.h"
+
+namespace peloton {
+namespace test {
+
+using namespace optimizer;
+
+const int N_ROW = 10000;
+
+class CostTests : public PelotonTest {};
+
+// tablename: test
+// database name: DEFAULT_DB_NAME
+void CreateAndLoadTable() {
+  TestingSQLUtil::ExecuteSQLQuery(
```

Comment:
This is because we want to have the control over the tuples we want to insert, so that we can know the basic ground truth of the costs and build some `EXPECT` statement to test our cost functions.